### ISSUE TITLE
Database Update Command Fails

### ISSUE THREAD

**Issue by stefanbenten** [nextcloud/server#8303]
### Steps to reproduce
1. Start Update from NC 12.0.5
2. Proceed with Web Based Updater
3. Let the Updater start

### Expected behaviour
I expect that the Table Update Commands proceed without an issue.

### Actual behaviour
The Updater stops with the following failure:

```
Doctrine\DBAL\Exception\DriverException: An exception occurred while executing ' UPDATE `oc_filecache` SET `mimetype` = ? WHERE `mimetype` <> ? AND `mimetype` <> ? AND `name` COLLATE utf8mb4_general_ci LIKE ? ' with params [false, 8, false, "%.tcx"]: SQLSTATE[22007]: Invalid datetime format: 1292 Truncated incorrect DOUBLE value: ''
```

### Server configuration

**Operating system**: Debian 9

**Web server:** Apache 2.4.25

**Database:** MariaDB 10.2.11 Cluster with maxscale Loadbalancer

**PHP version:** 7.0.19

**Nextcloud version:** 12.0.5

**Updated from an older Nextcloud/ownCloud or fresh install: Recent Updates proceed without any issues**

**Where did you install Nextcloud from: Download Package**

**Signing status:**
<details>
<summary>Signing status</summary>

```
Login as admin user into your Nextcloud and access 
http://example.com/index.php/settings/integrity/failed 
paste the results here.
```
</details>

**List of activated apps:**
<details>
<summary>App list</summary>

```
Enabled:
  - activity: 2.6.1
  - admin_notifications: 1.0.1
  - announcementcenter: 3.1.1
  - bruteforcesettings: 1.0.3
  - comments: 1.3.0
  - dav: 1.4.6
  - federatedfilesharing: 1.3.1
  - federation: 1.3.0
  - files: 1.8.0
  - files_pdfviewer: 1.2.0
  - files_sharing: 1.5.0
  - files_texteditor: 2.5.1
  - files_trashbin: 1.3.0
  - files_versions: 1.6.0
  - files_videoplayer: 1.2.0
  - firstrunwizard: 2.2.1
  - gallery: 18.0.0
  - logreader: 2.0.0
  - lookup_server_connector: 1.1.0
  - nextcloud_announcements: 1.2.0
  - notifications: 2.1.2
  - oauth2: 1.1.0
  - password_policy: 1.3.0
  - provisioning_api: 1.3.0
  - serverinfo: 1.3.0
  - sharebymail: 1.3.0
  - survey_client: 1.1.0
  - systemtags: 1.3.0
  - theming: 1.4.1
  - twofactor_backupcodes: 1.2.3
  - updatenotification: 1.3.0
  - user_ldap: 1.3.1
  - workflowengine: 1.3.0
Disabled:
  - admin_audit
  - encryption
  - files_external
  - quota_warning
  - user_external
```
</details>

**Nextcloud configuration:**
<details>
<summary>Config report</summary>

```
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "cloud.stefan-benten.de"
        ],
        "trusted_proxies": "***REMOVED SENSITIVE VALUE***",
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "overwrite.cli.url": "https:\/\/cloud.stefan-benten.de",
        "htaccess.RewriteBase": "\/",
        "memcache.local": "\\OC\\Memcache\\APCu",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "filelocking.enabled": "true",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 0,
            "timeout": 0
        },
        "dbtype": "mysql",
        "version": "12.0.5.3",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "ldapIgnoreNamingRules": false,
        "ldapProviderFactory": "\\OCA\\User_LDAP\\LDAPProviderFactory",
        "mail_smtpmode": "smtp",
        "mail_smtpauthtype": "LOGIN",
        "mail_smtpsecure": "tls",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "25",
        "maintenance": false,
        "theme": "",
        "loglevel": 1,
        "updater.secret": "***REMOVED SENSITIVE VALUE***"
    }
}
```
</details>

**Are you using external storage, if yes which one:** local

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory

#### LDAP configuration (delete this part if not used)
<details>
<summary>LDAP config</summary>

```
With access to your command line run e.g.:
sudo -u www-data php occ ldap:show-config
from within your Nextcloud installation folder

Without access to your command line download the data/owncloud.db to your local
computer or access your SQL server remotely and run the select query:
SELECT * FROM `oc_appconfig` WHERE `appid` = 'user_ldap';


Eventually replace sensitive data as the name/IP-address of your LDAP server or groups.
```
</details>

### Client configuration
**Browser:** Chrome 63

**Operating system:** Windows 10

#### Web server error log
<details>
<summary>Web server error log</summary>

```
Insert your webserver log here
```
</details>

#### Nextcloud log (data/nextcloud.log)
<details>
<summary>Nextcloud log</summary>

```
Insert your Nextcloud log here
```
</details>

#### Browser log
<details>
<summary>Browser log</summary>

```
Update vorbereiten
Log-Level auf "debug" gesetzt
Wartungsmodus eingeschaltet 
Das Datenbankschema wird aktualisiert
Datenbank aktualisiert
Suche nach einer Aktualisierung für die App "activity" im App-Store
App-Store auf Aktualalisierung für die App "activity" geprüft
Suche nach einer Aktualisierung für die App "admin_notifications" im App-Store
App-Store auf Aktualalisierung für die App "admin_notifications" geprüft
Suche nach einer Aktualisierung für die App "announcementcenter" im App-Store
App-Store auf Aktualalisierung für die App "announcementcenter" geprüft
Suche nach einer Aktualisierung für die App "bruteforcesettings" im App-Store
App-Store auf Aktualalisierung für die App "bruteforcesettings" geprüft
Suche nach einer Aktualisierung für die App "comments" im App-Store
App-Store auf Aktualalisierung für die App "comments" geprüft
Suche nach einer Aktualisierung für die App "dav" im App-Store
App-Store auf Aktualalisierung für die App "dav" geprüft
Suche nach einer Aktualisierung für die App "federatedfilesharing" im App-Store
App-Store auf Aktualalisierung für die App "federatedfilesharing" geprüft
Suche nach einer Aktualisierung für die App "federation" im App-Store
App-Store auf Aktualalisierung für die App "federation" geprüft
Suche nach einer Aktualisierung für die App "files" im App-Store
App-Store auf Aktualalisierung für die App "files" geprüft
Suche nach einer Aktualisierung für die App "files_pdfviewer" im App-Store
App-Store auf Aktualalisierung für die App "files_pdfviewer" geprüft
Suche nach einer Aktualisierung für die App "files_sharing" im App-Store
App-Store auf Aktualalisierung für die App "files_sharing" geprüft
Suche nach einer Aktualisierung für die App "files_texteditor" im App-Store
App-Store auf Aktualalisierung für die App "files_texteditor" geprüft
Suche nach einer Aktualisierung für die App "files_trashbin" im App-Store
App-Store auf Aktualalisierung für die App "files_trashbin" geprüft
Suche nach einer Aktualisierung für die App "files_versions" im App-Store
App-Store auf Aktualalisierung für die App "files_versions" geprüft
Suche nach einer Aktualisierung für die App "files_videoplayer" im App-Store
App-Store auf Aktualalisierung für die App "files_videoplayer" geprüft
Suche nach einer Aktualisierung für die App "firstrunwizard" im App-Store
App-Store auf Aktualalisierung für die App "firstrunwizard" geprüft
Suche nach einer Aktualisierung für die App "gallery" im App-Store
App-Store auf Aktualalisierung für die App "gallery" geprüft
Suche nach einer Aktualisierung für die App "logreader" im App-Store
App-Store auf Aktualalisierung für die App "logreader" geprüft
Suche nach einer Aktualisierung für die App "lookup_server_connector" im App-Store
App-Store auf Aktualalisierung für die App "lookup_server_connector" geprüft
Suche nach einer Aktualisierung für die App "nextcloud_announcements" im App-Store
App-Store auf Aktualalisierung für die App "nextcloud_announcements" geprüft
Suche nach einer Aktualisierung für die App "notifications" im App-Store
App-Store auf Aktualalisierung für die App "notifications" geprüft
Suche nach einer Aktualisierung für die App "oauth2" im App-Store
App-Store auf Aktualalisierung für die App "oauth2" geprüft
Suche nach einer Aktualisierung für die App "password_policy" im App-Store
App-Store auf Aktualalisierung für die App "password_policy" geprüft
Suche nach einer Aktualisierung für die App "provisioning_api" im App-Store
App-Store auf Aktualalisierung für die App "provisioning_api" geprüft
Suche nach einer Aktualisierung für die App "serverinfo" im App-Store
App-Store auf Aktualalisierung für die App "serverinfo" geprüft
Suche nach einer Aktualisierung für die App "sharebymail" im App-Store
App-Store auf Aktualalisierung für die App "sharebymail" geprüft
Suche nach einer Aktualisierung für die App "survey_client" im App-Store
App-Store auf Aktualalisierung für die App "survey_client" geprüft
Suche nach einer Aktualisierung für die App "systemtags" im App-Store
App-Store auf Aktualalisierung für die App "systemtags" geprüft
Suche nach einer Aktualisierung für die App "theming" im App-Store
App-Store auf Aktualalisierung für die App "theming" geprüft
Suche nach einer Aktualisierung für die App "twofactor_backupcodes" im App-Store
App-Store auf Aktualalisierung für die App "twofactor_backupcodes" geprüft
Suche nach einer Aktualisierung für die App "updatenotification" im App-Store
App-Store auf Aktualalisierung für die App "updatenotification" geprüft
Suche nach einer Aktualisierung für die App "user_ldap" im App-Store
App-Store auf Aktualalisierung für die App "user_ldap" geprüft
Suche nach einer Aktualisierung für die App "workflowengine" im App-Store
App-Store auf Aktualalisierung für die App "workflowengine" geprüft
Doctrine\DBAL\Exception\DriverException: An exception occurred while executing ' UPDATE `oc_filecache` SET `mimetype` = ? WHERE `mimetype` <> ? AND `mimetype` <> ? AND `name` COLLATE utf8mb4_general_ci LIKE ? ' with params [false, 8, false, "%.tcx"]: SQLSTATE[22007]: Invalid datetime format: 1292 Truncated incorrect DOUBLE value: ''
```
</details>

**Comment by kimbaxz** [comment]
Got the same error this morning. 
I've deployed a docker container with the version 12.X, test it for 5 minutes. and make a new build with the lastest version. Change the build into the docker-compose. And run it. 

I got the maintenance msg, change it with the config.php file, and launch the upgrade process, but got the same error msg.  

"Doctrine\DBAL\Exception\DriverException: An exception occurred while executing ' UPDATE `oc_filecache` SET `mimetype` = ? WHERE `mimetype` <> ? AND `mimetype` <> ? AND `name` COLLATE utf8mb4_general_ci LIKE ? ' with params [false, 5, false, "%.kml"]: SQLSTATE[22007]: Invalid datetime format: 1292 Truncated incorrect DOUBLE value: ''" 

Some news on this bug?
